### PR TITLE
Create Application Model

### LINKED ISSUE #8: [ENGINEERING] Create the application model
**What is the issue?**
Create a Application mongoose model using the [backend documentation](https://docs.google.com/document/d/1etcX9DOrmVatbwaTZ_kqpn9m8IiAPh_zeOQatQ5KPjM/edit?usp=sharing) with an application status sub document. 

**Additional context**
The model should be created in `/backend/src/models`, for reference take a look at the User model.

### PR DESCRIPTION
**Pull request by ulises0516**
## Changes

What changes did you make? Include screenshots if applicable, or explain how to view the changes.

- created application and applicationStatus models according to backend docs

## Testing

How did you confirm your changes work? (Automated tests, manual verification, etc.)

- verified models by crosschecking with the schema provided in the backend docs (manual verification)

## Tracking

Resolves #8

### CODE REVIEW

**Comment by ulises0516** [comment]
If you can make a request to make a change to the pr. I can add
` timestamps: { createdAt: 'dateCreated', updatedAt: 'dateModified' }` and get rid of the dateCreated and dateModified fields so that it will follow the backend docs. I am sorry about that!

File: backend/src/models/Application.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { InferSchemaType, Schema, model } from "mongoose";
+import { applicationStatusSchema } from "./ApplicationStatus";
+
+const applicationSchema = new Schema({
+  _id: {
+    type: String,
+    unique: true,
+    required: true,
+  },
```
Comment (kevinsun0904) [inline backend/src/models/Application.ts:9]: The `_id` field would be automatically generated by mongoose so I think we can remove this portion

File: backend/src/models/Application.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    default: [],
+  },
+  dateCreated: {
+    type: Date,
+    default: Date.now,
+  },
+  dateModified: {
+    type: Date,
+    default: Date.now,
+  },
```
Comment (kevinsun0904) [inline backend/src/models/Application.ts:41]: Apologies for the confusion, I meant for the `dateCreated` and `dateModified` fields to be the default `timestamps` fields (`createdAt` and `modifiedAt`) so feel free to change this to `timestamps`. I will make sure to update the backend doc! And also FYI I think you should be able to modify the PR by just simply pushing to the same branch in your remote repo without a request to change.

**Review by kevinsun0904 — CHANGES_REQUESTED** [review]
Overall very good! just a few minor changes

File: backend/src/models/ApplicationStatus.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  },
+});
+
+type ApplicationStatus = InferSchemaType<typeof applicationStatusSchema>;
+
+export { applicationStatusSchema };
+export default model<ApplicationStatus>(
+  "ApplicationStatus",
+  applicationStatusSchema,
+);
```
[the remark this answers is not in the document]
Comment (ulises0516) [reply backend/src/models/ApplicationStatus.ts:20]: You are right! I will make that change.

File: backend/src/models/Application.ts (hunk 1 above)
Comment (ulises0516) [reply backend/src/models/Application.ts:9]: Noted

**Review by kevinsun0904 — APPROVED** [review]
LGTM

**Pull request merged**